client/internal/httpsender_test.go

```diff
@@ -57,6 +87,33 @@ func TestHTTPSenderRetryForStatusTooManyRequests(t *testing.T) {
 	srv.Close()
 }
 
+func TestHTTPSenderRetryForStatusNotFound(t *testing.T) {
```

Comment:
I am not user I understand the purpose of the test. The name suggest that it verifies that sender tries on a status not found, but I don't see the code in this test that actually does the verification. I see a single sendRequestWithRetries() call and verification that it returns an error. Where is the retry part?

Reply:
I got this understanding based on your msg.
1. function name should be `TestHTTPSenderStatusNotFound` as this is exactly what it is testing and `retryfor` should be removed as it is not required.
2. Currently in `httpsender.go` I am sending response as `nil` and `error` for case `StatusNotFound`, so if exactly wants to check for StatusNotFound, then we have to send `HTTPresponse` too instead of nil, if that makes sense?